docs/docs/how-to/images-and-media/using-gatsby-plugin-image.md

```diff
@@ -0,0 +1,161 @@
+
+   ```jsx
+   import { StaticImage } from "gatsby-plugin-image"
+
+   export function Dino() {
+     return (
+       <StaticImage
+         src="../images/dino.png"
+         alt="A dinosaur"
+         placeholder="blurred"
+         layout="fixed"
+         width={200}
+         height={200}
+       />
+     )
+   }
+   ```
+
+### Dynamic images
+
+If you need to have dynamic images, such as if they are coming from a CMS, they are loaded via GraphQL and displayed using the `GatsbyImage` component.
+
+1. **Add the image to your page query**:
+
+   Any GraphQL File object that includes an image will have a `childImageSharp` field that you can use to query the image data. The exact data structure will vary according to your data source, but the syntax is like this:
+
+   ```graphql
+   query {
+     blogPost(id: { eq: $Id }) {
+       title
+       body
+       avatar {
+         childImageSharp {
+           gatsbyImageData(maxWidth: 200)
+         }
+       }
+     }
+   }
+   ```
+
+2. **Configure your image**:
+
+   You configure the image by passing arguments to the `gatsbyImageData` resolver. You can change the size and layout, as well as settings such as the type of placeholder used when lazy loading. There are also advanced image processing options available. You can find the full list of options in the API docs.
+
+   ```graphql
+   query {
+     blogPost(id: { eq: $Id }) {
+       title
+       body
+       author
+       avatar {
+         childImageSharp {
+           gatsbyImageData(
+             maxWidth: 200
+             placeholder: BLURRED
+             formats: [AUTO, WEBP, AVIF]
+           )
+         }
+       }
+     }
+   }
+   ```
+
+3. **Display the image:**
+
+   You can then use the `GatbsyImage` component to display the image on the page. The `getImage()` function is an optional helper to make your code easier to read. It takes a `File` and returns `file.childImageSharp.gatsbyImageData`, which can be passed to the `GatsbyImage` component.
+
+   ```jsx
+   import { GatsbyImage, getImage } from "gatsby-plugin-image"
+
+   function BlogPost({ data }) {
+     const image = getImage(data.blogPost.avatar)
+     return (
+       <section>
+         <h2>{data.blogPost.title}</h2>
+         <GatsbyImage image={image} alt={data.blogPost.author} />
+         <p>{data.blogPost.body}</p>
+       </section>
+     )
+   }
+   ```
+
+## Migrating
+
+If your site uses the old `gatsby-image` component, you can use a codemod to help you migrate to the new Gatsby Image components. This can update the code for most sites. To use the codemod, run this command in the root of your site:
+
+```shell
+npx gatsby-codemods gatsby-plugin-image
+```
+
+This will convert all GraphQL queries and components to use the new plugin. For more information see the full migration guide.
```

Comment:
Also assuming this will link to the migration guide?

Reply:
Yup